gym_ignition/base/robot/robot_baseframe.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import numpy as np
+from typing import Tuple
+from abc import ABC, abstractmethod
+
+# TODO: new exception for not existing joints? Add Exceptions in the docs?
+
+
+class RobotBaseFrame(ABC):
+    """
+    Robot base frame interface.
+
+    This interface provides methods to get and set quantities related to the robot base.
+    """
+
+    def __init__(self) -> None: ...
+
+    @abstractmethod
+    def set_base_frame(self, frame_name: str) -> bool:
+        """
+        Set the base frame of the robot. If this method is not called, the detection of
+        the base frame depends on the implementation.
+
+        Args:
+            frame_name: The name of the base frame.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def base_frame(self) -> str:
+        """
+        Get the name of the base frame.
+
+        Returns:
+            The name of the base frame.
+        """
+
+    @abstractmethod
+    def base_pose(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Returns the pose of the robot base.
+
+        The pose is composed of a position and a quaternion associated to the base frame,
+        both expressed in the the world inertial frame.
+
+        Returns:
+            A Tuple containing the position and orientation of the base:
+
+            - position: 3D array in the [x, y, z] form.
+            - orientation: a 4D array containing a quaternion in the [w, x, y, z] form.
+        """
+
+    @abstractmethod
+    def base_velocity(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Return the velocity of the robot base.
+
+        The velocity is composed of linear and angular velocities of the base wrt the
+        world inertial frame, both expressed in the world inertial frame.
```

Comment:
Written like this, I think most people will may confuse this with the inertial velocity of Featherstone (i.e. ${}^A\mathrm{v}\_{A,B}$ ), while I guess you mean something like the mixed velocity (but I am just guessing here), i.e. ${}^{B[A]} \mathrm{v}_{A,B}$ .

Reply:
If you can write LaTeX, I would write an minimal redacted version of our usual notation:  
* Describe what ${}^A o_B \in \mathbb{R}^3 and {}^A R_B \in \mathbb{R}^{3 \times 3}$ are 
* Define the skew symmetric operators vee and wedge, to go back and forth 3D vectors 
* Document that with "inear and angular velocities of the base wrt the world inertial frame" you mean: 
${}^A \dot{o}_B$ and  ${}^A \omega_{A,B} := ( {}^A \dot{R}_B {}^A R_B^T )^\vee$
Then in the code you can just refer to were you define the velocity as such. In this way, with a short amount of text, you precisely defined the convention that you use.

Reply:
Note that this is the text that we tipically copy paste in any "Notation" section of our papers, so we can copy from there also in this case as well.

Reply:
I opened https://github.com/robotology/gym-ignition/issues/37 as a reminder.

Reply:
Thanks, for the suggestions! :)